src/get_imports.rs

```diff
@@ -0,0 +1,215 @@
+use std::collections::{HashMap, HashSet};
+use std::iter::FromIterator;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use swc_common::errors::Handler;
+use swc_common::source_map::Pos;
+use swc_common::SourceFile;
+use swc_ecma_parser::Capturing;
+use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
+
+use crate::error::GetImportError;
+
+pub fn get_imports_map_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    get_imports_from_file(&file_path)
+}
+
+fn get_imports_from_file<'a>(
+    file_path: &'a PathBuf,
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    let path_string = match file_path.to_str() {
+        Some(path) => path,
+        None => {
+            return Err(GetImportError::PathError {
+                filepath: file_path.clone(),
+            })
+        }
+    };
+    let cm = Arc::<swc_common::SourceMap>::default();
+    let fm = match cm.load_file(Path::new(path_string)) {
+        Ok(f) => f,
+        Err(e) => {
+            return Err(GetImportError::FileDoesNotExist {
+                filepath: path_string.to_string(),
+                io_errors: vec![e],
+            });
+        }
+    };
+
+    let mut parser_errors: Vec<String> = Vec::new();
+    let dest_vector: Vec<u8> = Vec::new();
+    let dst = Box::new(dest_vector);
+    let handler = Handler::with_emitter_writer(dst, Some(cm.clone()));
+    let lexer = create_lexer(&fm);
+    let capturing = Capturing::new(lexer);
+    let mut parser = Parser::new_from(capturing);
+
+    let errors = parser.take_errors();
+
+    if !errors.is_empty() {
+        for error in errors {
+            let mut diagnostic = error.into_diagnostic(&handler);
+            parser_errors.push(diagnostic.message());
+            diagnostic.cancel();
+        }
+        return Err(GetImportError::ParseTsFileError {
+            filepath: path_string.to_string(),
+            parser_errors,
+        });
+    }
+
+    let ts_module = match parser.parse_typescript_module() {
+        Ok(module) => module,
+        Err(error) => {
+            let mut diagnostic = error.into_diagnostic(&handler);
+            parser_errors.push(diagnostic.message());
+            diagnostic.cancel();
+            return Err(GetImportError::ParseTsFileError {
+                filepath: path_string.to_string(),
+                parser_errors,
+            });
+        }
+    };
+
+    let imports_map = capture_imports_map(ts_module, fm);
+
+    return Ok(imports_map);
+}
+
+fn capture_imports_map(
+    ts_module: swc_ecma_ast::Module,
+    fm: Arc<SourceFile>,
+) -> HashMap<String, Option<HashSet<String>>> {
+    let mut imports_map: HashMap<String, Option<HashSet<String>>> = HashMap::new();
+    ts_module.body.iter().for_each(|node| {
```

Comment:
Missing logic here to capture imports from `import()` and `require()` expressions. Doesn't have to be part of this change, I filed #4 as a reminder to do it in a follow-up